File: velolabs_repos/skylock_ios/Skylock/Skylock/TheftHandler.swift
//
//  TheftHandler.swift
//  Ellipse
//
//  Created by Andre Green on 2/7/17.
//  Copyright © 2017 Andre Green. All rights reserved.
//

import Foundation

class TheftHandler: Accelerometer {
    override var maxPoints:Int {
        return 20
    }
    
    private var sensitivity:Float
    
    private let cutOff:Float = 195.73
    
    init(sensitivity: Float) {
        self.sensitivity = sensitivity
    }
    
    func set(sensitivity: Float) {
        self.sensitivity = sensitivity
    }
    
    override func shouldAlert() -> Bool {
        if points.count < maxPoints {
            return false
        }
        
        let stdDevs = allStdDevs()
        guard let x = stdDevs[.xValue], let y = stdDevs[.yValue], let z = stdDevs[.zValue] else {
            return false
        }
        
        if checkAxis(value: x) || checkAxis(value: y) || checkAxis(value: z) {
            print("Throwing theft alert with values: x: \(x) y: \(y) z: \(z)")
            points.removeAll()
            return true
        }
        
        return false
    }
    
    func checkAxis(value: Float) -> Bool {
        return value - 10.0 * (sensitivity - 0.5) > cutOff
    }
}

File: velolabs_repos/Skyfleet-Android/app/src/main/java/cc/skylock/skylock/bluetooth/TheftHandler.java
package cc.skylock.skylock.bluetooth;

import android.util.Log;

/**
 * Created by admin on 11/02/17.
 */

public class TheftHandler extends Accelerometer {
    String TAG = TheftHandler.class.getName();

    private float sensitivity;

    private double cutOff = 195.73;

    public TheftHandler(float sensitivity ) {
        this.sensitivity = sensitivity;
         super.maxPoints = 20;
    }

    public void set(float sensitivity ) {
        this.sensitivity = sensitivity;
    }

    @Override
    public boolean shouldAlert() {
        if(accelerometerValues.size() < maxPoints) {
            return false;
        }
        float[] stdDevs = allStdDevs();
        float x = stdDevs[0];
        float y = stdDevs[1];
        float z = stdDevs[2];

        if(checkAxis(x) || checkAxis(y) || checkAxis(z)){
            Log.i(TAG,"Throwing theft alert with values: x:"+x+",y:"+y+",z:"+z);
            accelerometerValues.clear();
            return true;
        }
        return false;
    }

    public boolean checkAxis(float value){
        return value + 10.0 * (sensitivity - 0.5) > cutOff;
    }
}
